crates/chain-state/src/in_memory.rs

```diff
@@ -837,13 +862,13 @@ pub enum NewCanonicalChain<N: NodePrimitives = EthPrimitives> {
     /// A simple append to the current canonical head
     Commit {
         /// all blocks that lead back to the canonical head
-        new: Vec<ExecutedBlock<N>>,
+        new: Vec<ExecutedBlockWithTrieUpdates<N>>,
     },
     /// A reorged chain consists of two chains that trace back to a shared ancestor block at which
     /// point they diverge.
     Reorg {
         /// All blocks of the _new_ chain
-        new: Vec<ExecutedBlock<N>>,
+        new: Vec<ExecutedBlockWithTrieUpdates<N>>,
         /// All blocks of the _old_ chain
         old: Vec<ExecutedBlock<N>>,
```

Comment:
this makes sense.

can we document why these are different block types?

Reply:
@klkvr this type is the common way users will encounter this change, so we should add a note about the old chain here as well, why trieupdates are missing for the reorged chain